lib/scholar/metrics.ex

```diff
@@ -331,6 +331,76 @@ defmodule Scholar.Metrics do
     Nx.divide(true_negatives, false_positives + true_negatives + 1.0e-16)
   end
 
+  @doc ~S"""
+  Calculates the confusion matrix.
+
+  ## Argument Shapes
+
+    * `y_true` - $\(d_0, d_1, ..., d_n\)$
+    * `y_pred` - $\(d_0, d_1, ..., d_n\)$
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([1.0, 1.0, 2.0, 2.0, 3.0, 3.0], type: {:f, 32})
+      iex> y_pred = Nx.tensor([1.0, 2.0, 1.0, 3.0, 3.0, 3.0], type: {:f, 32})
+      iex> Scholar.Metrics.confusion_matrix(y_true, y_pred)
+  """
+  def confusion_matrix(y_true, y_pred) do
+    {unique_label_atoms, unique_labels, num_classes} =
+      [y_true, y_pred]
+      |> Nx.concatenate()
+      |> Nx.to_flat_list()
+      |> Enum.reduce(%{}, fn label, acc ->
+        Map.update(acc, String.to_atom("#{label}"), label, fn _ -> label end)
```

Comment:
This probably will be removed upon refactoring to defn, but `String.to_atom` is a no-go because of atom table exhaustion

Reply:
I would also avoid any to_flat_list calls because they take data away from the device